File: HotDesk_SPICE/TMS_HotDesk/Library/Library.Root/Other/BusinessLogicBase.vb
Imports System.Threading

Namespace Other
    ''' <summary>
    ''' Handler the Common Business Logic Function
    ''' -------------------------------------------
    ''' C.C.Yeon    25 April 2011   Initial Version
    ''' </summary>
    ''' <remarks></remarks>
    Public MustInherit Class BusinessLogicBase



        Public Enum LanguagePack
            English = 0
            Malay = 1
        End Enum

        Public Shared ReadOnly Property Language() As LanguagePack
            Get
                Dim lp As LanguagePack = LanguagePack.English

                If Thread.CurrentThread.CurrentCulture.ToString.Equals("ms-MY") Then
                    lp = LanguagePack.Malay
                End If

                Return lp
            End Get
        End Property




        ''' <summary>
        ''' Max Quantity per Page
        ''' </summary>
        ''' <value></value>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Shared ReadOnly Property MaxQuantityPerPage() As Integer
            Get
                Return CInt(System.Configuration.ConfigurationManager.AppSettings("MaxRowPerPage").ToString)
            End Get
        End Property

        ''' <summary>
        ''' Generate and Caculate the Number 
        ''' </summary>
        Public Shared ReadOnly Property FromRowNo(ByVal PageNo As Integer) As Integer
            Get
                If PageNo = 1 Then
                    Return 1
                Else
                    Return ((PageNo - 1) * MaxQuantityPerPage) + 1
                End If
            End Get
        End Property

        ''' <summary>
        ''' Generate and Caculate the Number 
        ''' </summary>
        Public Shared ReadOnly Property ToRowNo(ByVal PageNo As Integer) As Integer
            Get
                If PageNo = 1 Then
                    Return MaxQuantityPerPage
                Else
                    Return ((PageNo - 1) * MaxQuantityPerPage) + MaxQuantityPerPage
                End If
            End Get
        End Property

    End Class
End Namespace

File: HotDesk_SPICE/Library/Library.Root/Other/BusinessLogicBase.cs
using System.Threading;

namespace Other
{
    /// <summary>
    /// Handler the Common Business Logic Function
    /// -------------------------------------------
    /// C.C.Yeon    25 April 2011   Initial Version
    /// </summary>
    public abstract class BusinessLogicBase
    {
        public enum LanguagePack
        {
            English = 0,
            Malay = 1
        }

        public static LanguagePack Language
        {
            get
            {
                LanguagePack lp = LanguagePack.English;

                if (Thread.CurrentThread.CurrentCulture.ToString().Equals("ms-MY"))
                {
                    lp = LanguagePack.Malay;
                }

                return lp;
            }
        }

        /// <summary>
        /// Max Quantity per Page
        /// </summary>
        public static int MaxQuantityPerPage
        {
            get
            {
                return (int)(System.Convert.ToInt32(System.Configuration.ConfigurationManager.AppSettings["MaxRowPerPage"].ToString()));
            }
        }

        /// <summary>
        /// Generate and Caculate the Number 
        /// </summary>
        public static int FromRowNo(int PageNo)
        {
            if (PageNo == 1)
            {
                return 1;
            }
            else
            {
                return ((PageNo - 1) * MaxQuantityPerPage) + 1;
            }
        }

        /// <summary>
        /// Generate and Caculate the Number 
        /// </summary>
        public static int ToRowNo(int PageNo)
        {
            if (PageNo == 1)
            {
                return MaxQuantityPerPage;
            }
            else
            {
                return ((PageNo - 1) * MaxQuantityPerPage) + MaxQuantityPerPage;
            }
        }
    }
}
